core/common/src/ConcurrentQuery.ts

```diff
@@ -46,6 +46,7 @@ export interface QueryLimit {
 /** @public */
 export interface QueryPropertyMetaData {
   className: string;
+  accessString: string;
```

Comment:
For an interface to be backwards compatible with previous versions, you cannot add a required property like this one. I assume that if this were optional then the interface would be backwards-compatible.

Reply:
yes, you would need to add it as optional

Reply:
> A change to a user-constructible type is breaking when: a required property is added to the type, since all of the user's existing constructions of the type will be incorrect 

https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgEIRgeyig3gWAChkTk4AuZAI00wBsI4QBuIgXyKNElkRQEEYPZAWKkK1WgyasxJKpRABXALZVosjoSIg4KiAGcADnzQZsEAKpGAJnEgiipZAzDIEmEAbBQlCMMCelOhYOMgAvCJklD5KKGyanIS6+samgjzWdg6izq7unt6+-oEglBnQEVESsfGJhEA

Reply:
What can we do if we don't want `accessString` to be an optional property? Should we update the package version?

Reply:
https://www.itwinjs.org/learning/api-support-policies/#breaking-changes

If you want to avoid making a breaking change, id recommend adding a new API, deprecating the old one, and making sure in your code you handle the possibility of using either new or deprecated api. 

i think the easiest way forward is to add an optional property

Reply:
While these are interfaces we are the only implementers.  We use the interface to have the same API in the backend and frontend.  

If we make it required it could break on the frontend if the server is running a version from before this change.  Not sure, but seems possible.

I think our best option is to make this property optional instead of trying to do something 'smart'.  For iTwin.js version 5.x we can make this a non-optional property.  If we take this route we should create an issue now to make non-optional and tag for 5.x.

We should also take this opportunity to better document this interface.  We should either document each property in detail or link to docs on how things like jsonName and name are generated.

Reply:
I've updated `accessString` in the `QueryPropertyMetaData` interface to be an optional property. I've also documented how other properties are generated.

Reply:
Created an issue for iTwin.js 5.0 milestone https://github.com/iTwin/itwinjs-core/issues/6548